lib/puppet/http/service/puppetserver.rb

```diff
@@ -26,13 +26,24 @@ def initialize(client, session, server, port)
   # @api private
   #
   def get_simple_status(ssl_context: nil)
-    response = @client.get(
-      with_base_url("/status/v1/simple/master"),
-      headers: add_puppet_headers({}),
-      options: {ssl_context: ssl_context}
-    )
+    request_path = "/status/v1/simple/master"
```

Comment:
I don't think we can remove `.../master` in the next 2 years, as there will always be versions of 5.5.x and 6.(0-18) that still call it. So I'm thinking we should try `.../server` and fallback to `.../master`. It does mean an "extra round trip".... so I'm inclined to only make this change in Puppet 7, where we know the server almost certainly supports the new endpoint.

Reply:
I know this wasn't directed at me, but can I say what I _think_ is the situation here for my own clarity?

The agent should work with any server of that major version stream. ie, Agent 6.x should work with any Server 6.x and should default to pinging the `master` endpoint and "fall forward(?)" to `server`.

Server 6.x (which still support Puppet Agent 3.x) should have `server` added as a service in addition to `master` (and 🙇 thank you NW so much for opening that PR). Since the Server supports Agents multiple majors behind it it will probably have to continue having a `master` status service for another 6+ years and several majors.

In Agent 7.x we should default to pinging the `server` service and fall back to `master` to make upgrades/roll outs as uninteresting as possible for folks. In Agent 8.x we can drop the `master` fall back.

Reply:
Thanks for writing that up @justinstoller! I agree with everything except this part:

> Agent 6.x... and "fall forward(?)" to `server`

If the `master` endpoint isn't going away until well after puppet6 is EOL, then I'm thinking falling forward should never be necessary. So I think this PR can be closed, but we should move forward with the one targeting `main` which defaults to `server` and falls back to `master`.

Reply:
I agree that if we're going to keep around the `master` endpoint in server 6.x, this changes are not needed.